subprojects/core/src/main/java/org/gradle/internal/vfs/AbstractFileWatcherRegistryFactory.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.vfs;
+
+import net.rubygrapefruit.platform.file.FileWatchEvent;
+import net.rubygrapefruit.platform.file.FileWatcher;
+import org.gradle.internal.UncheckedException;
+import org.gradle.internal.vfs.watch.FileWatcherRegistry;
+import org.gradle.internal.vfs.watch.FileWatcherRegistryFactory;
+import org.gradle.internal.vfs.watch.FileWatcherUpdater;
+
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+
+public abstract class AbstractFileWatcherRegistryFactory implements FileWatcherRegistryFactory {
+    @Override
+    public FileWatcherRegistry startWatcher(FileWatcherRegistry.ChangeHandler handler) {
```

Comment:
Should this method be called `createFileWatcherRegistry()` or `startWatcherRegistry()` or somesuch?

Reply:
Renamed to `createFileWatcherRegistry`.